src/machine/machine_atsamd21_usb.go

```diff
@@ -20,19 +20,24 @@ const (
 	usb_DEVICE_PCKSIZE_MULTI_PACKET_SIZE_Pos  = 14
 	usb_DEVICE_PCKSIZE_MULTI_PACKET_SIZE_Mask = 0x3FFF
 
-	NumberOfUSBEndpoints = 8
+	NumberOfUSBEndpoints = 6
 )
 
 var (
-	endPoints = []uint32{
+	inEndpoints = []uint32{
```

Comment:
You could consider changing the code to something like:

```suggestion
	inEndpoints = [...]uint32{
```

That makes it a fixed-size array instead of a slice.

If you do that, you can probably also make `NumberOfUSBEndpoints` constant (possibly even moving it to shared code):

```go
NumberOfUSBEndpoints = max(len(inEndpoints), len(outEndpoints))
```

Reply:
Ah, good idea. That reminds me, this PR fixes the reason that endpoints were moved from machine/usb.go out into each platform, would be a good time to undo that.